src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+public abstract class Figure {
+    private String color;
+
+    public Figure(String color) {
+        this.color = color;
+    }
+
+    public abstract double getArea();
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    public void draw() {
+        moreInfo();
+        System.out.println("Area: " + getArea() + " sq. units");
+        System.out.println("Color: " + color);
+        System.out.println();
+    }
+
+    protected abstract void moreInfo();
+}

```

Comment:
Don't use abstract classes to set behavior for classes. In this case, it looks like the `draw()` method is common for all figures, which suggests it could be implemented in a non-abstract class or interface with default methods if you need to share common behavior.